nixos/modules/system/boot/systemd-nspawn.nix

```diff
@@ -42,8 +42,113 @@ let
+      options = {
+        source = mkOption {
+          type = str;
+          default = "";
+          description = "Optional mount source path. If omitted, the specified path will be mounted from the host to the same path in the container.";
+        };
+        mountOption = mkOption {
+          type = str;
+          default = "";
+          description = ''
+            Optional mount options. See
+            <link xlink:href="url">https://www.freedesktop.org/software/systemd/man/systemd-nspawn.html#Mount%20Options</link>
+            for more info.
+          '';
+        };
+      };
+    });
+    default = {};
+    example = { "/nix/store" = { source = "/nix/store"; mountOption = ""; }; };
+    description = "Mount host path into container.";
+  };
+
+  fromBindMounts = mapAttrsToList (dest: { source, mountOption, ... }:
+    if mountOption != "" then "${source}:${dest}:${mountOption}"
+    else if source != "" then "${source}:${dest}"
+    else toString dest);
+
+  fromTmpfsMounts = map (dest:
+    if builtins.isString dest then dest
+    else "${dest.path}:${dest.mountOption}");
+
+  fromInaccessibles = map toString;
+
+  overlayMounts = mkOption {
+    type = with types; attrsOf (listOf str);
+    default = {};
+    example = { "/var" = [ "+/readonly-var" "" ]; };
+    description = "Mount host path into the container.";
+  };
+
+  fromOverlayMounts = mapAttrsToList (dest: sources:
+    if length sources == 0 then throw "Overlay mount source path list is empty."
+    else "${concatStringsSep ":" sources}:${dest}");
+
+  fromVeths = mapAttrsToList (container: host: "${host}:${container}");
+
   instanceOptions = {
-    options = sharedOptions // {
+    options = {
+
+      environment = mkOption {
+        default = {};
+        example = { PATH = "/run/current-system/sw/bin"; };
+        type = with types; attrsOf (nullOr (oneOf [ int str path package ]));
+        description = "Environment variables passed to the container's init.";
+      };
+
+      bind = bindMounts;
+      bindReadOnly = bindMounts;
+
+      temporaryFileSystem = mkOption {
+        type = with types; listOf
+          (oneOf
+            [ str
+              (submodule {
+                options = {
+                  path = mkOption {
+                    type = str;
+                    default = "";
+                    description = "mount destination";
+                  };
+                  mountOption = mkOption {
+                    type = str;
+                    default = "";
+                    description = ''
+                      Mount options. See
+                      <link xlink:href="url">https://www.freedesktop.org/software/systemd/man/systemd-nspawn.html#Mount%20Options</link>
+                      for more info.
+                    '';
+                  };
+                };
+              })
+            ]
+          );
+        default = [];
+        example = [ "/tmp" { path = "/var/tmp"; mountOption = ""; } ];
```

Comment:
Please use `literalExample` for examples that contain a list or an attr-path for better formatting in the manual.